diff --git a/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.simd.hpp b/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.simd.hpp
index cea726a374..dd117a6684 100644
--- a/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.simd.hpp
+++ b/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.simd.hpp
@@ -1353,7 +1353,7 @@ cv::dnn::Winofunc getWinofunc_F32()
 //==============================================================================
 
 // FP16, currently, only ARMv8 may support it
-#if defined(CV_CPU_COMPILE_NEON_FP16) && CV_CPU_COMPILE_NEON_FP16
+#if defined(CV_CPU_COMPILE_NEON_FP16) && CV_CPU_COMPILE_NEON_FP16 && defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
 
 #undef T4x4
 #define T4x4(a, b, c, d, tr0, tr1) \
